[Ready] Icebox Moonstation Mine Time

Whoneedspacee opened the pull request:
## About The Pull Request

Video: https://www.youtube.com/watch?v=kX-mH4k8b6k&t=

Thanks to @wesoda25 and @KomradeSpectre for making ruins for ice moon

Adds icebox, it's boxstation on top of the ice moon mine, lavaland and space ruins don't spawn on this map. There are 3 z levels, the top level with the station and some ruins and mobs, the middle level with some ruins, and the bottom level with megafauna and more dangerous ruins. No need for a gravity generator. Bombcap multiplier does not work on the station level because bluespace blah blah blah.

Icemoon has it's own unique atmos quite like lavaland, you need a winter coat, explorer suit, equal or better to withstand the cold temperature. It has a couple unique mobs and spawners for them with polar bears and wolfs, wolfs run away at low health and polar bears attack faster. There are also now snow legions for ice moon.

Refactors some parts of the weather system, adds ZTRAIT booleans for specific weather rather than targeting areas like the mining ZTRAIT, allowing for more fine control. You can now also make weather target outdoor areas automatically, making weather much easier to setup for a zlevel so you don't have to specify every area it targets.

Refactors many parts of the tunnel generation system for lavaland, you can now have areas specify whether tunnels, flora, mobs, or megafauna can be allowed in them. This makes it much easier to map in ruins and make sure generation won't mess with them. Also fixes a bug where tunnels were checking areas around the object creating the tunnel and not the front of the tunnel, which let tunnels move through areas it shouldn't have. You can also have a variable tunnel width if you desire now.

Rocks can now be mined for volcanic ash which can allow for glass creation on places that don't have basalt or sand turf.

Refactors a part of ruin spawning which allows for ruins to be placed in multiple areas. This makes multi-z ruin spawning much more possible since before you needed the top and bottom z levels to have the same area.

The wabbajack proc has an argument to specify the type of transformation now, (transform into a human, not a xenomorph, etc)

Fixes a bug where asteroid mobs never showed their aggro icon state.

Syndicate shuttle can now land on mineral turfs.

Adds minetype "none" so some stations can have their own custom minetype on the map json itself and not have errors logged.

Adds new blood drunk miner type that can be spawned from destroyed nether portals.

Adds anywhere ruins which are ruins that can be placed anywhere without messing with atmosphere and other things (generally ruins with turf passover everywhere, maybe some ruins could be converted to this if not already).

Nuclear Operative lockers now spawn with a diamond drill so they can drill through mineral turfs, especially needed for if they fall off the top z level.

Blobs can no longer place their starting core on a no blob allowed area, they already couldn't swap their core to such a position so it makes sense that they shouldn't be able to place themselves on it by default.

## Why It's Good For The Game

It's pretty much a normal station except with a few turns that I think might make the game more interesting. There's planetary atmosphere so there's the potential for gas mining that is less feasible on the station. Power is going to be more of a concern since the supermatter can't automagically super cool everything with space, though using freezers for cooling is still an option if you want a normal SM setup, solars and the TEG might be more viable options.

The potential for multi-z exploration is interesting though I'm sure there will be some issues there because... it's multi-z. Chucking people down z-levels instead of into space will also be hilarious (and possibly more dangerous).

Overall I'm not really sure how it will go, but hey it's a votable map so if people don't like it the statistics will speak for themselves, at the very least something interesting for a bit or for admin events.

## Changelog
:cl:
add: Icebox Station has been added, explore the depths of the Ice Moon with boxstation right on top of it, don't forget your coat and internals!
add: You can now mine rock flora into volcanic ash to make glass.
/:cl:

Whoneedspacee committed:
there was an attempt

Whoneedspacee committed:
adds ice moon map

adds config options for choosing mining maps

areas now have options that affect random generation on them

weather now has an option to not target indoor areas

adds base for icemoon ruins

turns many 1's into booleans

adds ice chasms

adds ztraits for weather stuff

Whoneedspacee committed:
fixes up ice ruins to be ice moon compatible

adds lower z level to the ice moon mine

ice chasms now have a smoothed icon

megafauna only spawn in the underground portion of ice moon now

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
openspace fixes

Whoneedspacee committed:
adds new areas and underground specific ruins

KomradeSpectre committed:
Adds the abandoned village ruin for Ice Moon

Whoneedspacee committed:
Merge pull request #1 from KomradeSpectre/icemoon

Adds the abandoned village ruin for Ice Moon

Whoneedspacee committed:
adds abandoned village ruin and fixes some area and tile stuff to work with ice moon

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
random mining maps are no longer picked in the config

you can no longer change the mining map before setup is complete

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
adds above and below ground ruin as a test

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
adds debug functionality for ice moon ruins

fixes a bug where multiz was impossible unless there were shared areas because of whitelisted areas

adds multiple whitelist areas for ruin placement in a list

all underground areas are now outdoors

underground plasma lava rivers are now only spawned in unexplored areas so ruins don't get destroyed by them

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
adds unique spawners to icemoon to replace tendrils

adds 2 new mobs to icemoon the polar bear and wolf

adds a clothing flag for shoes that dont slip on ice

modifies mining site to be ready except for the boss

adds the ability for tunnel width to be a specification, ice moon can have 1 tile wide tunnels and rarely 2 wide tunnels

adds a no caves mineral for ice moon ruins

Whoneedspacee committed:
wolves no longer run into lava or chasms

bears now enrage sooner

bear spawners can now actually be created

adds base for ice moon atmospherics

adds base for a new boss and achievements / score

really bad coder icon for ice boots added

Whoneedspacee committed:
ice moon now has it's own planetary atmos! (thank god)

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
new frost miner stuff

megafauna recovery time can now be tailored to have different cooldowns to ranged or melee attacks
by default sets both of them, with two arguments the first is ranged cooldown and the second is melee cooldown

Whoneedspacee committed:
converts a bunch of lavaland maps to be compatible with ice moon

adds tendrils from lavaland to the bottom of ice moon with the other megafauna

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
updates mob drops for ice moon mobs

updates config entry default value for ice moon

updates ice moon map to have new gulag

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
updates station parallax for icemoon

removes extra lavaland ported ruins

updates to demonic frost miner

adds ladder to icemoon map near station

Whoneedspacee committed:
updates ice moon map to have the ladder inside the mining base

plant flora dont spawn on the lower z level of ice moon now

you get sand from mining rocks now

buncha demonic frost miner changes holy cow

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

wesoda25 committed:
adds the buried library

wesoda25 committed:
improves map, fixes comment

Whoneedspacee committed:
make this crap work with the station cause i have to lol it is pretty cool though and now i understand the mapping crap better

Whoneedspacee committed:
updates ice ruin blacklist

Whoneedspacee committed:
Merge pull request #2 from wesoda25/wanshitong-take-2

Adds the Buried Library Ruin

Whoneedspacee committed:
ice moon stop being too cold thanks

icebox changes, now actually has its own z level that can spawn ruins and such instead of just being openspace

tunnel width is now not stupid

Whoneedspacee committed:
Merge branch 'icemoon' of https://github.com/Whoneedspacee/tgstation into icemoon

Whoneedspacee committed:
fixes a demonic frost miner bug where orbs werent exploding after death

atmosphere temp is now correct

fixes library ruin for icemoon

polar bear range is now fixed, only aggros in a short range and follows further than 3 tiles

reverts necropolis chest change

wesoda25 committed:
Adds a new ruin: bathouse, barebones edition

Whoneedspacee committed:
Merge pull request #3 from wesoda25/bathhouse

Adds a new ruin: bathouse

Whoneedspacee committed:
adds many new ruins

fixes a bug with tunnel creation that caused it to enter no tunnel allowed areas

finishes the frost miner and adds loot to him

you can now force a wabbajack option

Whoneedspacee committed:
Merge branch 'icemoon' of https://github.com/Whoneedspacee/tgstation into icemoon

Whoneedspacee committed:
main z level now always has gravity for ice box

Whoneedspacee committed:
hopefully the hot springs arent the buggiest thing in the world anymore

protects areas near the station from having openspace

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
icebox takes less players

fixes the ladder positioning on every icemoon map

Whoneedspacee committed:
snowstorms now only occur on the station level

fixes some issues with the icebox z level having openspace over the lower level mining base

Bawhoppen commented:
Couldn't tell in the video since there's music, but do the storms make noise? If so, you may want to disable that, since I promise you that it will put your map on the fast-track to being unpopular.

Donpedrito commented:
If the planet has atmosphere, how does it work for miners' PKAs? Are the lower z-levels low-pressure?

Whoneedspacee commented:
Every z-level has the same planetary atmos, it's the same thing as lavaland your kinetic accelerators will always work at maximum efficiency.

wesoda25 commented:
W

Time-Green commented:
Who needs space?

Dennok commented:
Space can be added and accessible via teleport or shuttles.
But for start it interesting to place spaceless map.

HomocidlePancakes commented:
Do spessruins still spawn?

Dennok commented:
> Do spessruins still spawn?

As i read now. No. 
No space, no spaceruins.

Tharcoonvagh commented:
Testmerge?

Michael-Ashfield commented:
This seems like it would radically affect the SM cooling loop, is the planetary atmos cold enough to support normal SM setups?

Dennok commented:
> This seems like it would radically affect the SM cooling loop, is the planetary atmos cold enough to support normal SM setups?

As i memorize atmosfere has lower border of not damaging temperature. There can be broblems with cooling to extreme low temp but hot gas can be cooled almost like in space. 
I think, here is no big difference when we cool 1000K in pipe by 7K on space or 150K in cold atmosfere. Maybe this is better to cooling in atmosfere then in space. Need testing.

> How would this affect ops, presumably their ship now blasts an area clear where it lands?

Shuttle can lands where it designed, replace all turfs to shuttle turfs on landing.
Can have issues on manual dock designation.

> Can Blob spread on planet tiles, if so it would make blob unstoppable as it could set up 400 tiles away and win before it's found.

Blob counts only in station areas, as i know.

Ty-the-Smonk commented:
Planet station took years to give up on... now some gamer comes in with box on a cooler lava land in like a week

Ty-the-Smonk commented:
Also space is gonna need a new nukie ending animation since it’s a planet now

Tharcoonvagh commented:
How the hell would nukies get on the station

Tharcoonvagh commented:
You gotta love windows facing an ice wall

I still absolutely love it tho

spookydonut commented on code/datums/map_config.dm:
```diff
@@ -140,4 +140,4 @@
 
 /datum/map_config/proc/MakeNextMap()
-	return config_filename == "data/next_map.json" || fcopy(config_filename, "data/next_map.json")
+	return config_filename == "data/next_map.json" || fcopy(config_filename, "data/next_map.json")
```
```suggestion
	return config_filename == "data/next_map.json" || fcopy(config_filename, "data/next_map.json")

```

Whoneedspacee replied:
pfft my text editor and not just a terrible habit i have

spookydonut commented on code/datums/ruins/lavaland.dm:
```diff
@@ -262,4 +262,4 @@
 	always_place = TRUE
 	allow_duplicates = FALSE
-	always_spawn_with = list(/datum/map_template/ruin/space/kinggoatarena = PLACE_ISOLATED)
+	always_spawn_with = list(/datum/map_template/ruin/space/kinggoatarena = PLACE_ISOLATED)
```
```suggestion
	always_spawn_with = list(/datum/map_template/ruin/space/kinggoatarena = PLACE_ISOLATED)

```

spookydonut commented on code/game/objects/structures/lavaland/necropolis_tendril.dm:
```diff
@@ -94,4 +97,4 @@ GLOBAL_LIST_INIT(tendrils, list())
 		if(!T.density)
 			T.TerraformTurf(/turf/open/chasm/lavaland, /turf/open/chasm/lavaland, flags = CHANGETURF_INHERIT_AIR)
-	qdel(src)
+	qdel(src)
```
```suggestion
	qdel(src)

```

spookydonut commented on code/game/turfs/closed/minerals.dm:
```diff
@@ -568,4 +766,4 @@
 
 /turf/closed/mineral/strong/ex_act(severity, target)
-	return
+	return
```
```suggestion
	return

```

spookydonut commented on code/game/turfs/open/chasm.dm:
```diff
@@ -96,4 +105,4 @@
 	underlay_appearance.icon = 'icons/turf/floors.dmi'
 	underlay_appearance.icon_state = "dirt"
-	return TRUE
+	return TRUE
```
```suggestion
	return TRUE

```

spookydonut commented on code/game/turfs/open/floor/plating/asteroid.dm:
```diff
@@ -366,4 +437,4 @@ GLOBAL_LIST_INIT(megafauna_spawn_list, list(/mob/living/simple_animal/hostile/me
 /turf/open/floor/plating/asteroid/snow/atmosphere
 	initial_gas_mix = FROZEN_ATMOS
-	planetary_atmos = FALSE
+	planetary_atmos = FALSE
```
```suggestion
	planetary_atmos = FALSE

```

spookydonut commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,303 @@
+/obj/item/pickaxe/drill/jackhammer/demonic/Initialize()
+	..()
+	AddComponent(/datum/component/knockback, 4, TRUE)
+	AddComponent(/datum/component/lifesteal, 5)
```
```suggestion
	AddComponent(/datum/component/lifesteal, 5)

```

spookydonut commented on code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:
```diff
@@ -218,4 +220,4 @@
 	. = ..()
 	if(. > 0 && stat == CONSCIOUS)
-		Retaliate()
+		Retaliate()
```
```suggestion
		Retaliate()

```

spookydonut commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/polarbear.dm:
```diff
@@ -0,0 +1,54 @@
+	move_force = MOVE_FORCE_DEFAULT
+	move_resist = MOVE_RESIST_DEFAULT
+	pull_force = PULL_FORCE_DEFAULT
+	..(gibbed)
```
```suggestion
	..(gibbed)

```

spookydonut commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/wolf.dm:
```diff
@@ -0,0 +1,54 @@
+	. = ..()
+	if(target == null)
+		adjustHealth(-maxHealth*0.025)
+		retreat_message_said = FALSE
```
```suggestion
		retreat_message_said = FALSE

```

spookydonut commented on code/modules/ruins/icemoonruin_code/hotsprings.dm:
```diff
@@ -0,0 +1,44 @@
+					qdel(J)
+				var/datum/job/J = pick(valid_jobs)
+				J.equip(H, FALSE, TRUE, TRUE)
+				SSjob.SendToLateJoin(H)
```
```suggestion
				SSjob.SendToLateJoin(H)

```

spookydonut reviewed (changes requested):
time to fix your text editor bud, it's removing newlines from the end of files

Learner10 commented:
> 
> 
> Also space is gonna need a new nukie ending animation since it’s a planet now

Just clip in the Fichina base explosion from Star Fox. I'm sure that'd get a laugh or two.

Whoneedspacee committed:
Update code/datums/map_config.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/datums/ruins/lavaland.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/game/objects/structures/lavaland/necropolis_tendril.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/game/turfs/closed/minerals.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/game/turfs/open/chasm.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/game/turfs/open/floor/plating/asteroid.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/modules/mob/living/simple_animal/hostile/mining_mobs/polarbear.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/modules/mob/living/simple_animal/hostile/mining_mobs/wolf.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
Update code/modules/ruins/icemoonruin_code/hotsprings.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

Whoneedspacee committed:
fixes areas on the lust ruin

polar bears now drop a goliath plate equivalent

wolfs now drop a watcher sinew equivalent

adds snow legions

adds a crusher trophy for the demonic frost miner that prevents movement

fixes a bug with asteroid mobs where the aggro icon would never show

adds ice to the snow cave tunnel generation

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
Merge branch 'icemoon' of https://github.com/Whoneedspacee/tgstation into icemoon

Whoneedspacee commented:
> Testmerge?

yes

spookydonut commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,345 @@
+	if(!owner.stat)
+		to_chat(owner, "<span class='notice'>The cube melts!</span>")
+	owner.cut_overlay(cube)
+	UnregisterSignal(owner, COMSIG_MOVABLE_PRE_MOVE)
```
```suggestion
	UnregisterSignal(owner, COMSIG_MOVABLE_PRE_MOVE)

```

spookydonut commented on code/controllers/configuration/entries/game_options.dm:
```diff
@@ -389,4 +394,4 @@
 	min_val = 0
 
-/datum/config_entry/flag/dynamic_config_enabled
+/datum/config_entry/flag/dynamic_config_enabled
```
```suggestion
/datum/config_entry/flag/dynamic_config_enabled

```

spookydonut reviewed (changes requested):
same problem still

Whoneedspacee commented:
look i tried give me some credit it only happened on that one and the other one was preexisting

Whoneedspacee committed:
starts work on the wendigo megafauna

Qustinnus commented:
we need more than just the nukie vid right, we have a bunch of video animations that show a space station

Qustinnus commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/wolf.dm:
```diff
@@ -0,0 +1,56 @@
+
+/mob/living/simple_animal/hostile/asteroid/wolf/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(health <= maxHealth*0.1)
```
ditto

Qustinnus commented on code/modules/ruins/icemoonruin_code/hotsprings.dm:
```diff
@@ -0,0 +1,44 @@
+
+/turf/open/water/cursed_spring/Entered(atom/movable/thing, atom/oldLoc)
+	. = ..()
+	if(isliving(thing))
```
early returns

Qustinnus commented on code/__DEFINES/achievements.dm:
```diff
@@ -42,6 +43,7 @@
 #define BOSS_MEDAL_KINGGOAT   "King Goat Killer"
 
 #define BOSS_MEDAL_MINER_CRUSHER	  	"Blood-drunk Miner Crusher"
+#define BOSS_MEDAL_FROSTMINER_CRUSHER	"Demonic-frost Miner Crusher"
```
keep the whitespace formatting and indent this like the rest

Whoneedspacee replied:
doesn't look like this when I look on my local so don't know what to say

Qustinnus commented on code/__DEFINES/atmospherics.dm:
```diff
@@ -258,6 +258,7 @@
 
 //ATMOS MIX IDS
 #define LAVALAND_DEFAULT_ATMOS		"LAVALAND_ATMOS"
+#define ICEMOON_DEFAULT_ATMOS		"ICEMOON_ATMOS"
```
ditto

Whoneedspacee replied:
ditto

Qustinnus commented on code/game/objects/items/kitchen.dm:
```diff
@@ -140,6 +141,23 @@
+
+/obj/item/kitchen/knife/bloodletter/attack(mob/living/M, mob/living/carbon/user)
+	. =..()
+	if(istype(M) && (M.mob_biotypes & MOB_ORGANIC))
```
early return

Pumpkinoe commented:
how do nukies land if there's rock in the way???

Dennok commented:
> how do nukies land if there's rock in the way???

The predicted land site dont see on rocks, it penetrate it on land.

Whoneedspacee committed:
replaces snowball machine gun with more miner useful content

adds gentle var for knockback to prevent stunning on hit

polar bears move slower now

new icon for phase 2 demonic miner now

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
fixes lust ruin areas

fixes whitespace

fixes nesting issues

adds underground specific mobs

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
name / location fixes

demonic frost miner doesnt stun on knockback now

ice demons move faster

Whoneedspacee committed:
fixes icebox having the wrong baseturf

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

TheMidnightRose commented:
Is this close to testmerge ready?

Whoneedspacee commented:
tbh I’ve been focusing more on yogstation since they actually merged it but yes it should be ready, in bed right now will clear up merge conflicts tommorow

Whoneedspacee committed:
adds achievement defines to the wendigo

snowstorm for the lower z level again

adds a new helper for ranged target turfs

theres now a specific subtype for rivers

fixes a bug where ice and spawners spawned with rocks and other flora on top of them

adds indestructible ice rock turf

fixes a bug with ice demons teleport distance being incorrect

adds the start of wendigos attacks

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee commented:
go ahead and testmerge if you want though the wendigo boss is unfinished and drops no loot, pretty easy though since he only has one move.

ice demons also don't drop their loot yet.

Pumpkinoe commented:
Do the fauna have cruncher trophies yet?

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,68 @@
+
+/obj/structure/spawner/ice_moon/deconstruct(disassembled)
+	destroy_message()
+	. = ..()
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/mapping/ruins.dm:
```diff
@@ -11,8 +11,15 @@
+			if(check.flags_1 & NO_RUINS_1)
 				valid = FALSE
+			else
+				valid = FALSE // set to false before we check
```
```suggestion
			valid = FALSE // set to false before we check
			if(!(check.flags_1 & NO_RUINS_1))
```

Rohesie commented on code/modules/mob/living/carbon/human/human_movement.dm:
```diff
@@ -12,13 +12,18 @@
+		if(shoes && istype(shoes, /obj/item/clothing))
+			var/obj/item/clothing/CS = shoes
+			if (CS.clothing_flags & NOSLIP_ICE)
+				return 0
```
```suggestion
				return FALSE
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,354 @@
+
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/Move()
+	if(!enraging)
+		. = ..()
```
```suggestion
/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/Goto(target, delay, minimum_distance)
	if(enraging)
		return
	return ..()

/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/MoveToTarget(list/possible_targets)
	if(enraging)
		return
	return ..()

/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/Move()
	if(enraging)
		return
	return ..()
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,354 @@
+			new /obj/item/clothing/shoes/winterboots/ice_boots/speedy(T)
+		if(3)
+			new /obj/item/pickaxe/drill/jackhammer/demonic(T)
+	. = ..()
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,354 @@
+	toolspeed = 0
+
+/obj/item/pickaxe/drill/jackhammer/demonic/Initialize()
+	..()
```
```suggestion
	. = ..()
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/wendigo.dm:
```diff
@@ -0,0 +1,162 @@
+	if(!can_move)
+		return
+	stored_move_dirs |= direct
+	. = ..()
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/wendigo.dm:
```diff
@@ -0,0 +1,162 @@
+		exit.id = "wendigo arena exit"
+		exit.add_atom_colour(COLOR_RED_LIGHT, ADMIN_COLOUR_PRIORITY)
+		exit.set_light(20, 1, LIGHT_COLOR_RED)
+		. = ..()
```
```suggestion
	var/obj/effect/portal/permanent/one_way/exit = new /obj/effect/portal/permanent/one_way(starting)
	exit.id = "wendigo arena exit"
	exit.add_atom_colour(COLOR_RED_LIGHT, ADMIN_COLOUR_PRIORITY)
	exit.set_light(20, 1, LIGHT_COLOR_RED)
	return ..()
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/ice demon.dm:
```diff
@@ -0,0 +1,72 @@
+	var/turf/end = pick(possible_ends)
+	do_teleport(src, end, 0,  channel=TELEPORT_CHANNEL_BLUESPACE, forced = TRUE)
+	SLEEP_CHECK_DEATH(8)
+	. = ..()
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/ice demon.dm:
```diff
@@ -0,0 +1,72 @@
+	move_force = MOVE_FORCE_DEFAULT
+	move_resist = MOVE_RESIST_DEFAULT
+	pull_force = PULL_FORCE_DEFAULT
+	..(gibbed)
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/ice whelp.dm:
```diff
@@ -0,0 +1,52 @@
+	move_force = MOVE_FORCE_DEFAULT
+	move_resist = MOVE_RESIST_DEFAULT
+	pull_force = PULL_FORCE_DEFAULT
+	..(gibbed)
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/polarbear.dm:
```diff
@@ -0,0 +1,56 @@
+	move_force = MOVE_FORCE_DEFAULT
+	move_resist = MOVE_RESIST_DEFAULT
+	pull_force = PULL_FORCE_DEFAULT
+	..(gibbed)
```
```suggestion
	return ..()
```

Rohesie commented on code/modules/mapping/ruins.dm:
```diff
@@ -52,7 +59,7 @@
 
 
-/proc/seedRuins(list/z_levels = null, budget = 0, whitelist = /area/space, list/potentialRuins)
+/proc/seedRuins(list/z_levels = null, budget = 0, whitelist = list(/area/space), list/potentialRuins)
```
Why is `whitelist` a list var? Despite its name it seems to actually be an area variable in functionality.

Whoneedspacee replied:
It is the areas that a ruin can be placed, it is a list so you can have multiple areas that a ruin can be allowed to be placed in, this is necessary for ruin generation in z levels with multiple areas or multi-z ruins where the ruin is placed on a top and bottom z level with different areas.

Whoneedspacee committed:
Apply suggestions from code review

Co-Authored-By: Rohesie <rohesie@gmail.com>

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,68 @@
+/obj/structure/spawner/ice_moon/deconstruct(disassembled)
+	destroy_message()
+	return ..()
+
+/obj/structure/spawner/ice_moon/proc/destroy_message()
+	visible_message("<span class='userdanger'>[src] collapses, sealing everything inside!</span>")
+
+/obj/structure/spawner/ice_moon/proc/clear_rock()
+	for(var/turf/F in RANGE_TURFS(2, src))
```
New procs need documentation.

Rohesie replied:
Why is this resolved?

Rohesie commented on code/modules/mapping/ruins.dm:
```diff
@@ -11,8 +11,15 @@
+						valid = TRUE
+						break
+			if(!valid)
 				break
```
```suggestion
			valid = FALSE // set to false before we check
			if(check.flags_1 & NO_RUINS_1)
				break
			for(var/type in allowed_areas)
				if(istype(new_area, type)) // it's at least one of our types so it's whitelisted
					valid = TRUE
					break
		if(!valid)
			break
```

Rohesie replied:
This logic is a bit funky. Are you sure it's correct?

Whoneedspacee replied:
pretty sure havent seen any ruins generate over the station nor has yogstation

Whoneedspacee committed:
Update code/modules/mapping/ruins.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
Fixes a dumb bug with ruins from a webedit review

The syndicate shuttle can now land on mineral turfs

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
the final commit, all that im going to change after this is documentation for procs lol

makes both ice moon z levels below the station underground, and makes it so they generate rivers of their baseturf

Whoneedspacee committed:
adds nice animation to wendigo scream

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:
```diff
@@ -69,20 +69,19 @@
+		if(!elimination)	//used so the achievment only occurs for the last legion to die.
+			grant_achievement(achievement_type, score_achievement_type, crusher_kill, force_grant)
+			SSblackbox.record_feedback("tally", tab, 1, "[initial(name)]")
+	..()
```
```suggestion
	return ..()
```

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,68 @@
+
+/obj/structure/spawner/ice_moon/polarbear
+	max_mobs = 1
+	spawn_time = 600 //60 seconds
```
```suggestion
	spawn_time = 60 SECONDS
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,357 @@
+
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/OpenFire()
+	check_enraged()
+	projectile_speed_multiplier = 1 + enraged / 2 // ranges from normal to 2x speed
```
```suggestion
	projectile_speed_multiplier = 1 + enraged * 0.5 // ranges from normal to 2x speed
```
Prefer multiplications over divisions.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/wendigo.dm:
```diff
@@ -0,0 +1,202 @@
+	attack_action_types = list(/datum/action/innate/megafauna_attack/heavy_stomp,
+							   /datum/action/innate/megafauna_attack/teleport,
+							   /datum/action/innate/megafauna_attack/disorienting_scream)
+	var/turf/starting
+	var/stomp_range = 1
+	var/stored_move_dirs = 0
+	var/can_move = TRUE
```
Please document new variables.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/wendigo.dm:
```diff
@@ -0,0 +1,202 @@
+	invocation = "RAAAAAAAAWR!"
+	convert_damage = FALSE
+
+	shapeshift_type = /mob/living/simple_animal/hostile/asteroid/polarbear
```
```suggestion
	shapeshift_type = /mob/living/simple_animal/hostile/asteroid/polarbear

```

Whoneedspacee committed:
small fixes

Whoneedspacee committed:
finishes autodoccing everything i could find

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
ok im done for real now

Whoneedspacee commented:
ok its done for real now

TheMidnightRose commented:
Now to hopefully see it testmerged in the next few days.

Rohesie commented on code/game/objects/items/kitchen.dm:
```diff
@@ -144,6 +145,25 @@
+	/// Bleed stacks applied when an organic mob target is hit
+	var/bleed_stacks_per_hit = 3
+
+/obj/item/kitchen/knife/bloodletter/attack(mob/living/M, mob/living/carbon/user)
+	. = ..()
+	if(!istype(M) || !(M.mob_biotypes & MOB_ORGANIC))
+		return
+	var/datum/status_effect/stacking/saw_bleed/bloodletting/B = M.has_status_effect(/datum/status_effect/stacking/saw_bleed/bloodletting)
+	if(!B)
+		M.apply_status_effect(/datum/status_effect/stacking/saw_bleed/bloodletting, bleed_stacks_per_hit)
+	else
+		B.add_stacks(bleed_stacks_per_hit)
```
This will happen even if the attacker is pacifist and the attack doesn't happen.
Is that intended?

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,172 @@
+  */
+/obj/structure/spawner/ice_moon/proc/drop_loot()
+	// types with probability chance to appear out of 100
+	var/ore_types = list(/obj/item/stack/ore/uranium = 50,
+						 /obj/item/stack/ore/iron = 100,
+						 /obj/item/stack/ore/plasma = 75,
+						 /obj/item/stack/ore/silver = 50,
+						 /obj/item/stack/ore/gold = 50,
+						 /obj/item/stack/ore/diamond = 25,
+						 /obj/item/stack/ore/bananium = 5,
+						 /obj/item/stack/ore/titanium = 75)
+	for(var/type in ore_types)
```
Turn this into a global list.

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,172 @@
+	playsound(loc,'sound/effects/tendril_destroyed.ogg', 200, FALSE, 50, TRUE, TRUE)
+	visible_message("<span class='boldannounce'>[src] begins to collapse, cutting it off from this world!</span>")
+	animate(src, transform = matrix().Scale(0, 1), alpha = 50, time = 50)
+	sleep(50)
+	visible_message("<span class='warning'>Something slips out of [src]!</span>")
```
Use a callback.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,367 @@
+
+/// Called when the orb is exploding, shoots out projectiles
+/obj/projectile/frost_orb/proc/orb_explosion(projectile_speed_multiplier)
+	var/list/spread = list(0, 60, 120, 180, 240, 300)
+	for(var/angle in spread)
```
```suggestion
	for(var/i in 0 to 5)
	var/angle = i * 60
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,367 @@
+/obj/item/clothing/shoes/winterboots/ice_boots/speedy
+	name = "cursed ice hiking boots"
+	desc = "A pair of winter boots contractually made by a devil, they cannot be taken off once put on."
+	slowdown = SHOES_SLOWDOWN - 1
```
Make a define for it, it's more maintainable.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,367 @@
+	duration = 25
+	status_type = STATUS_EFFECT_REFRESH
+	alert_type = /obj/screen/alert/status_effect/ice_block_talisman
+	var/icon/cube
```
Document.

Rohesie replied:
Be sure to qdel it on `Destroy()`

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:
```diff
@@ -69,21 +82,21 @@
-				grant_achievement(achievement_type, score_achievement_type, crusher_kill, force_grant)
-				SSblackbox.record_feedback("tally", tab, 1, "[initial(name)]")
-		..()
+	var/datum/status_effect/crusher_damage/C = has_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
+	var/crusher_kill = FALSE
+	if(C && crusher_loot && C.total_damage >= maxHealth * 0.6)
```
```suggestion
	var/datum/status_effect/crusher_damage/crusher_dmg = has_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
	var/crusher_kill = FALSE
	if(crusher_dmg && crusher_loot && crusher_dmg.total_damage >= maxHealth * 0.6)
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/ice demon.dm:
```diff
@@ -0,0 +1,77 @@
+	SLEEP_CHECK_DEATH(8)
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/ice_demon/Life()
+	. = ..()
+	if(target == null)
+		adjustHealth(-maxHealth*0.025)
```
Check if the return is truthy, which would mean it's dead.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/ice whelp.dm:
```diff
@@ -0,0 +1,56 @@
+	var/list/burn_turfs = getline(src, T) - get_turf(src)
+	dragon_fire_line(src, burn_turfs)
+
+/mob/living/simple_animal/hostile/asteroid/ice_whelp/Life()
+	. = ..()
+	if(target == null)
+		adjustHealth(-maxHealth*0.025)
```
Check for a truthy return, meaning a dead mob.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/polarbear.dm:
```diff
@@ -0,0 +1,58 @@
+		aggressive_message_said = TRUE
+	rapid_melee = 2
+
+/mob/living/simple_animal/hostile/asteroid/polarbear/Life()
+	. = ..()
+	if(target == null)
+		adjustHealth(-maxHealth*0.025)
+		aggressive_message_said = FALSE
```
Check for a truthy return.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/wolf.dm:
```diff
@@ -0,0 +1,59 @@
+		retreat_message_said = TRUE
+	retreat_distance = 30
+
+/mob/living/simple_animal/hostile/asteroid/wolf/Life()
+	. = ..()
+	if(target == null)
+		adjustHealth(-maxHealth*0.025)
+		retreat_message_said = FALSE
```
Check for a truthy return.

Rohesie commented on code/modules/ruins/icemoonruin_code/hotsprings.dm:
```diff
@@ -0,0 +1,47 @@
+	var/mob/living/L = thing
+	if(!L.client)
+		return
+	if(L.mind in GLOB.cursed_minds)
+		return
+	GLOB.cursed_minds += L.mind
```
Is this list being cleared on mind destruction?

Rohesie replied:
You could also use an associative list for faster lookup:
```suggestion
	if(GLOB.cursed_minds[L.mind])
		return
	GLOB.cursed_minds[L.mind] = L.mind
```

Whoneedspacee committed:
adds anywhere ruins

does review stuff

Whoneedspacee commented:
this is good for testmerge i think

Rohesie commented on code/datums/ruins/icemoon.dm:
```diff
@@ -0,0 +1,106 @@
+	Seem very intent on research and individual liberty, and also geology-based naming?"
+	prefix = "_maps/RandomRuins/AnywhereRuins/"
+	suffix = "golem_ship.dmm"
+	allow_duplicates = FALSE
```
```suggestion
	allow_duplicates = FALSE

```

Rohesie commented on code/game/objects/items/kitchen.dm:
```diff
@@ -144,6 +145,25 @@
+	/// Bleed stacks applied when an organic mob target is hit
+	var/bleed_stacks_per_hit = 3
+
+/obj/item/kitchen/knife/bloodletter/attack(mob/living/M, mob/living/carbon/user)
+	. = ..()
+	if(!. || !istype(M) || !(M.mob_biotypes & MOB_ORGANIC))
+		return
+	var/datum/status_effect/stacking/saw_bleed/bloodletting/B = M.has_status_effect(/datum/status_effect/stacking/saw_bleed/bloodletting)
+	if(!B)
+		M.apply_status_effect(/datum/status_effect/stacking/saw_bleed/bloodletting, bleed_stacks_per_hit)
+	else
+		B.add_stacks(bleed_stacks_per_hit)
```
Are you sure this will ever happen?
I don't see a `return TRUE` in `/obj/item/proc/attack`

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,179 @@
+  */
+/obj/structure/spawner/ice_moon/proc/destroy_effect()
+	playsound(loc,'sound/effects/explosionfar.ogg', 200, TRUE)
+	visible_message("<span class='boldannounce'>[src] collapses, sealing everything inside!</span>")
+	visible_message("<span class='warning'>Ores fall out of the cave as it is destroyed!</span>")
```
```suggestion
	visible_message("<span class='boldannounce'>[src] collapses, sealing everything inside!</span>\n<span class='warning'>Ores fall out of the cave as it is destroyed!</span>")
```

Whoneedspacee replied:
why?

Rohesie replied:
Same result, a single proc called.

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,179 @@
+	playsound(loc,'sound/effects/tendril_destroyed.ogg', 200, FALSE, 50, TRUE, TRUE)
+	visible_message("<span class='boldannounce'>[src] begins to collapse, cutting it off from this world!</span>")
+	animate(src, transform = matrix().Scale(0, 1), alpha = 50, time = 50)
+	addtimer(CALLBACK(src, .proc/collapse), 50)
```
```suggestion
	addtimer(CALLBACK(src, .proc/collapse), 5 SECONDS)
```

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,179 @@
+	. = ..()
+	playsound(loc,'sound/effects/tendril_destroyed.ogg', 200, FALSE, 50, TRUE, TRUE)
+	visible_message("<span class='boldannounce'>[src] begins to collapse, cutting it off from this world!</span>")
+	animate(src, transform = matrix().Scale(0, 1), alpha = 50, time = 50)
```
```suggestion
	animate(src, transform = matrix().Scale(0, 1), alpha = 50, time = 5 SECONDS)
```

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,179 @@
+		if(27)
+			new /obj/item/book/granter/spell/sacredflame(loc)
+		if(28)
+			new /mob/living/simple_animal/hostile/megafauna/blood_drunk_miner/doom(loc)
```
```suggestion
			new /mob/living/simple_animal/hostile/megafauna/blood_drunk_miner/doom(loc)

```
Linter.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,368 @@
+	loot = list(/obj/effect/decal/remains/plasma)
+	wander = FALSE
+	del_on_death = TRUE
+	blood_volume = BLOOD_VOLUME_NORMAL
+	/// Modifies the speed of the projectiles the demonic frost miner shoots out
+	var/projectile_speed_multiplier = 1
+	/// If the demonic frost miner is in its enraged state
+	var/enraged = FALSE
+	/// If the demonic frost miner is currently transforming to its enraged state
+	var/enraging = FALSE
+	achievement_type = /datum/award/achievement/boss/demonic_miner_kill
```
Put the new variables at the bottom, under the redefined ones, for clarity.

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,368 @@
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/ex_act(severity, target)
+	adjustBruteLoss(30 * severity - 120)
+	visible_message("<span class='danger'>[src] absorbs the explosion!</span>", "<span class='userdanger'>You absorb the explosion!</span>")
+	return
```
```suggestion
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,368 @@
+		var/turf/endloc = get_turf(target)
+		if(!endloc)
+			break
+		var/obj/projectile/frost_orb/P = new /obj/projectile/frost_orb(startloc)
```
```suggestion
		var/obj/projectile/frost_orb/P = new(startloc)
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,368 @@
+		var/turf/endloc = get_turf(original)
+		if(!startloc || !endloc)
+			break
+		var/obj/projectile/P = new /obj/projectile/ice_blast(startloc)
```
```suggestion
		var/obj/projectile/ice_blast/P = new(startloc)
```

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,368 @@
+		if(!startloc || !endloc)
+			break
+		var/obj/projectile/P = new /obj/projectile/ice_blast(startloc)
+		P.speed /= projectile_speed_multiplier
```
Could you turn this into a multiplication?

Whoneedspacee replied:
you want *= 1/projectile_speed_multiplier?

Rohesie replied:
No, just change the logic as to use a multiplication.
For example, doing `projectile_speed_multiplier = 1 - enraged * 0.5`
Instead of dividing by two, you are multiplying by 0.5

Rohesie commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,368 @@
+		if(!endloc)
+			break
+		var/obj/projectile/P = new /obj/projectile/snowball(startloc)
+		P.speed /= projectile_speed_multiplier
```
And this one as well?

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/wolf.dm:
```diff
@@ -0,0 +1,60 @@
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/wolf/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(health > maxHealth*0.1)
+		retreat_distance = initial(retreat_distance)
+		return
+	if(!retreat_message_said && target)
+		visible_message("<span class='danger'>The [name] tries to flee from [target]!</span>")
+		retreat_message_said = TRUE
+	retreat_distance = 30
```
Couldn't this message happen right after the mob has died?

Whoneedspacee replied:
its supposed to happen before the mob dies??

Rohesie replied:
`adjustHealth` calls the parent, which calls `updatehealth()` which calls `update_stat()` is that not correct?
Which means by the time of your message on the mob trying to flee it is possibly `stat == DEAD`

Rohesie commented on code/modules/ruins/icemoonruin_code/hotsprings.dm:
```diff
@@ -0,0 +1,47 @@
+GLOBAL_LIST_EMPTY(cursed_minds)
```
Are minds removing themselves from this list on deletion?

Whoneedspacee replied:
no probably not

Rohesie replied:
Make sure that happens to avoid issues with garbage collection.

Whoneedspacee replied:
no idea what to do with this or what a null reference key in an association list would do

Time-Green replied:
The garbage collector wont kill off objects if they’re referenced somewhere, like here. Just remove them from the list once they get qdelled. You could probably add a signal to look for it and remove it when the time comes

Whoneedspacee replied:
even if they're hard deleted? strange

Whoneedspacee committed:
review stuff

Whoneedspacee committed:
ok it removes the stuff now

Whoneedspacee commented:
for realsies now

Time-Green reviewed (commented):
This doesnt remove it, it just changes the value the mind is associated with. 

Example: list(Mind = null)

Just do GLOB.cursed_minds -= M if you want it out of the list

Whoneedspacee committed:
fix removal

Rohesie commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/wolf.dm:
```diff
@@ -44,7 +44,7 @@
 /mob/living/simple_animal/hostile/asteroid/wolf/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
 	. = ..()
-	if(health > maxHealth*0.1)
+	if(stat == DEAD || health > maxHealth*0.1)
```
```suggestion
	if(stat != CONSCIOUS || health > maxHealth*0.1)
```

Rohesie commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,178 @@
+  */
+/obj/structure/spawner/ice_moon/proc/clear_rock()
+	for(var/turf/F in RANGE_TURFS(2, src))
+		if(abs(src.x - F.x) + abs(src.y - F.y) > 3)
```
```suggestion
		if(abs(x - F.x) + abs(y - F.y) > 3)
```

Rohesie commented:
This PR is pretty huge and definitely will need testing.

Arianya commented:
Testmerged on Sybil

Rohesie commented:
Some ruins don't mesh right with the ice map:
![5Bl3Tbi](https://user-images.githubusercontent.com/42041276/78849560-33b50c00-79eb-11ea-9f84-62ecab568a28.png)
![QScoJZe](https://user-images.githubusercontent.com/42041276/78849604-51827100-79eb-11ea-8300-f64a609092f7.png)

Whoneedspacee commented:
intentional but yeah stupid

ill try to make em like the whats it the uhh biodome whatever stuff

saprasam commented:
it's the same shitty box everybody knows but it has a built in meteor shield, literally no meteors ever injured the station at all even with a catastrophic meteor raid happening

Qustinnus commented:
making all ruins into bio domes is silly

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/polarbear.dm:
```diff
@@ -0,0 +1,59 @@
+/mob/living/simple_animal/hostile/asteroid/polarbear
```
it's a shame that these guys can't wear bear armor

maybe add that in later?

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/mining_mobs/ice whelp.dm:
```diff
@@ -0,0 +1,57 @@
+	butcher_results = list(/obj/item/stack/ore/diamond = 3, /obj/item/stack/sheet/sinew = 2, /obj/item/stack/sheet/bone = 10, /obj/item/stack/sheet/animalhide/ashdrake = 1)
+	loot = list()
+	crusher_loot = /obj/item/crusher_trophy/tail_spike
+	deathmessage = "collapses on it's side."
```
*its side

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,367 @@
+
+/obj/item/clothing/shoes/winterboots/ice_boots/speedy/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT)
```
er, can you put these on if you pick them up from the ground, or will they be stuck to your hand?

Whoneedspacee replied:
you can put them on but they will be stuck to your hand same as eye of god

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:
```diff
@@ -0,0 +1,367 @@
+
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner
+	name = "demonic-frost miner"
+	desc = "An extremely geared miner, driven crazy or possessed by the demonic forces here, either way a terrifying enemy."
```
maybe use "well-geared" instead of "geared"?

ATH1909 commented on code/game/objects/structures/icemoon/cave_entrance.dm:
```diff
@@ -0,0 +1,178 @@
+		if(14)
+			new /obj/item/disk/design_disk/modkit_disc/bounty(loc)
+		if(15)
+			new /obj/item/ship_in_a_bottle(loc)
```
does the ice moon have ice rivers or something?

if not, consider replacing this loot drop with something else

Whoneedspacee replied:
plasma rivers are on the bottom z level

ATH1909 commented on code/datums/components/knockback.dm:
```diff
@@ -41,4 +58,4 @@
 		throw_distance *= -1
 	var/atom/throw_target = get_edge_target_turf(throwee, throw_dir)
-	throwee.safe_throw_at(throw_target, throw_distance, 1, thrower)
+	throwee.safe_throw_at(throw_target, throw_distance, 1, thrower, gentle = throw_gentle)
```
sad

ATH1909 commented on code/datums/achievements/boss_achievements.dm:
```diff
@@ -62,12 +67,22 @@
 
+/datum/award/achievement/boss/demonic_miner_crusher
+	name = "Demonic-frost Miner Crusher"
+	desc = "Definitely harder than the Blood-drunk Miner."
```
*Blood-Drunk Miner

ATH1909 replied:
and apply these changes up above as well, I guess

ATH1909 commented on code/datums/achievements/boss_achievements.dm:
```diff
@@ -62,12 +67,22 @@
+	database_id = BOSS_MEDAL_WENDIGO
+
 /datum/award/achievement/boss/blood_miner_crusher
 	name = "Blood-drunk Miner Crusher"
```
*Blood-Drunk Miner

ATH1909 commented on code/datums/achievements/boss_achievements.dm:
```diff
@@ -62,12 +67,22 @@
 	icon = "miner"
 
+/datum/award/achievement/boss/demonic_miner_crusher
+	name = "Demonic-frost Miner Crusher"
```
Capitalize frost? Idk, this name seems weird.

ATH1909 commented on _maps/RandomRuins/IceRuins/icemoon_underground_wrath.dmm:
```diff
@@ -0,0 +1,267 @@
+/turf/closed/mineral/random/snow/no_caves,
+/area/icemoon/underground/explored)
+"e" = (
+/obj/structure/spawner/nether,
```
If you'd like a loot idea for this wrath ruin, perhaps a pair of gloves that let you rip apart (or butcher, for simplemobs) corpses with your bare hands? Because you hate hate hate HATE HATE HATE THEM SO MUCH DIE DIE DIE AND NEVER COME BACK I'LL RIP YOU TO ATOMS SO I NEVER HAVE TO SEE YOUR SMUG FACE AGAIN YOU PIECE OF SHIT RAAAAAAAAAAAAAAAAAAAAAAAGH

Whoneedspacee replied:
oh wow that's actually way better than the trash idea i was planning thanks

cool miner gear like that is the best

ATH1909 commented on _maps/RandomRuins/IceRuins/icemoon_surface_lust.dmm:
```diff
@@ -0,0 +1,171 @@
+/area/icemoon/surface/outdoors)
+"f" = (
+/obj/structure/falsewall/plastitanium,
+/obj/structure/sign/poster/contraband/lusty_xenomorph,
```
put an actual lusty xenomorph maid in here or no balls

Whoneedspacee replied:
the joke is that it's the lust ruin but it only has to do with lust as in lusting after an object like the greentext

ATH1909 replied:
ahhhh

Detective-Google commented on code/game/turfs/closed/minerals.dm:
```diff
@@ -214,14 +234,44 @@
 
+/turf/closed/mineral/random/volcanic/more_caves
+	mineralSpawnChanceList = list(
+		/obj/item/stack/ore/uranium = 5, /obj/item/stack/ore/diamond = 1, /obj/item/stack/ore/gold = 10, /obj/item/stack/ore/titanium = 11,
```
the rest of these use the turfs (i.e. /turf/closed/mineral/uranium), is it a mistake to use the item stacks here?

Whoneedspacee replied:
should be, someone made a pr a while back that changed these to be ore stacks instead

only reason my mineral turfs are turf paths is because i want the turfs to be different if they have ores

Whoneedspacee commented:
new pr
